File: 面试题18-删除链表的节点.py
class Solution:
    def deleteNode(self, head: ListNode, val: int) -> ListNode:
        if head.val == val:
            head = head.next
        
        else:
            cur_node = head
            while cur_node.next.val != val:
                cur_node = cur_node.next
            
            cur_node.next = cur_node.next.next

        return head

File: 面试题18-删除链表的节点.cpp
class Solution {
public:
    ListNode* deleteNode(ListNode* head, int val) {
        ListNode* dummy_node = new ListNode(-1);
        dummy_node->next = head;
        ListNode* cur_node = dummy_node;

        while(cur_node->next->val != val)
            cur_node = cur_node->next;

        cur_node->next = cur_node->next->next;

        return dummy_node->next;
    }
};
